app/tut/ga/nuair-a.md

```diff
@@ -0,0 +1,489 @@
+        méid := méideanna[j]
+
+        >-- Glan an stáitse.
+        glan@stáitse()
+
+        >-- Tarraing an ciorcal i lár an stáitse.
+        ciorcal@stáitse(fad_x@stáitse / 2, fad_y@stáitse / 2, méid)
+
+        >-- Codail ar feadh tamall beag.
+        codladh(400)
+    }
+}
+}}}
+
+# Stop! {#stad}
+
+Anois déanfaimid iarracht ríomhchlár nua a chruthú [[a ligeann don úsáideoir|that lets the user]]
+liosta téacs a chlóscríobh go dtí go gclóscríobhann siad "Stop", ansin scríobhfaidh an ríomhchlár an
+liosta ar ais ar an gconsól. Féach ar an GIF seo:
+
+![Ag clóscríobh liosta téacs](assets/ag-closcriobh-liosta-teacs.gif)
+
+Is féidir linn an gníomh `léigh`{.setanta} a úsáid chun [[ionchur an úsáideora|user input]] a fháil.
+Tosaímis le lúb `nuair-a fíor`{.setanta} agus feicimid níos déanaí
+[[conas éalú ón lúb|how to escape the loop]].
+
+```{.setanta .numberLines}
+>-- Cruthaigh liosta folamh.
+liosta := []
+
+nuair-a fíor {
+    >-- Faigh téacs ón úsáideoir.
+    téacs := léigh()
+
+    >-- Cuir an téacs leis an liosta.
+    liosta += [téacs]
+}
+```
+
+Má ritear an ríomhchlár sin, leanfaidh sé ar aghaidh go deo ag léigh téacs ón gconsól. Ba mhaith
+linn scoir ón lúb nuair a scríobhann an úsáideoir "Stop". Conas is féidir linn é sin a dhéanamh?
+
+## DIY
+
+Má úsáidimid coinníoll éigin [[in ionad|instead of]] `fíor`{.setanta}, beidh muid ábalta an lúb a
+stopadh. Cruthaímis athróg nua `stop` leis an luach `bréag`{.setanta} agus bainimis úsáid as an
+athróg sin mar coinníoll an lúibe mar seo:
+
+```{.setanta .numberLines}
+>-- Cruthaigh liosta folamh.
+liosta := []
+
+stop := bréag
+
+nuair-a !stop {
+    >-- Faigh téacs ón úsáideoir.
+    téacs := léigh()
+
+    >-- Cuir an téacs leis an liosta.
+    liosta += [téacs]
+}
+```
+
+Anois leanfaidh an lúb ag rith nuair atá `stop` cothrom le `bréag`{.setanta}, agus stopfaidh sé má
+tá sé cothrom lé `fíor`{.setanta}. Anois nuair a faighimid téacs ón úsáideoir, is féidir linn an
+athróg `stop` a athrú go `fíor`{.setanta} má tá an téacs cothrom le `"Stop"`{.setanta}.
+
+```{.setanta .numberLines}
+>-- Cruthaigh liosta folamh.
+liosta := []
+
+stop := bréag
+
+nuair-a !stop {
+    >-- Faigh téacs ón úsáideoir.
+    téacs := léigh()
+
+    má téacs == "Stop" {
+        stop = fíor
+    } nó {
+        >-- Cuir an téacs leis an liosta.
+        liosta += [téacs]
+    }
+}
+```
+
+Anois cuirimis líne ag an deireadh chun an liosta a scríobh amach agus beidh ríomhchlár ceart
+againn.
+
+{{{
+>-- Cruthaigh liosta folamh.
+liosta := []
+
+stop := bréag
+
+nuair-a !stop {
+    >-- Faigh téacs ón úsáideoir.
+    téacs := léigh()
+
+    má téacs == "Stop" {
+        stop = fíor
+    } nó {
+        >-- Cuir an téacs leis an liosta.
+        liosta += [téacs]
+    }
+}
+scríobh(liosta)
+}}}
+
+Bain triail as! Rith an ríomhchlár agus scríobh isteach cúpla rudaí difriúla, ansin scríobh "Stop"
+agus ba chóir duit liosta na rudaí a scríobh tú a fheiceáil sa chonsól.
+
+## An bealach níos fearr
```

Comment:
Bealach níos fearr nó An bealach is fearr

Reply:
Déanta grma